gapis/api/vulkan/frame_loop.go

```diff
@@ -371,6 +377,22 @@ func (f *frameLoop) Transform(ctx context.Context, cmdId api.CmdID, cmd api.Cmd,
 			f.loopTerminated = true
 			log.D(ctx, "FrameLoop: end loop at frame %v cmdId %v, cmd is %v.", f.frameNum, cmdId, cmd)
 
+			// This command is the last in the loop so lets add it to the captured commands so we don't need to special case it.
+			f.capturedLoopCmds = append(f.capturedLoopCmds, cmd)
+			f.capturedLoopCmdIds = append(f.capturedLoopCmdIds, cmdId)
+
+			// If we are looping zero times we can just drop the commands inside the loop.
+			if(f.loopCount == 0) {
+				f.capturedLoopCmds = make([]api.Cmd, 0)
+				f.capturedLoopCmdIds = make([]api.CmdID, 0)
+				return
+			}
+
+			// Some things we're going to need for the next work...
+			apiState := GetState(out.State())
+			stateBuilder := apiState.newStateBuilder(ctx, newTransformerOutput(out))
+			defer stateBuilder.ta.Dispose()
+
```

Comment:
I think a small comment giving the big picture would be nice here, something like:

```
Frame looping is achieved with a payload structure like this:
1. <pre-loop>
2. loop body
3. state rewind
4. loop body
5. jump to 3
6. <post-loop>
This is to enable a smooth transition to <post-loop> when the conditional jump is not taken.
```